### PR TITLE
Decrease logic in ProzesskopieForm

### PR DESCRIPTION
**Pull request by Beacze**

### CODE REVIEW

File: Kitodo/src/main/java/de/sub/goobi/forms/ProzesskopieForm.java [hunk 1]
```diff
@@ -2013,4 +739,20 @@ public boolean isCalendarButtonShowing() {
     public Fileformat getFileformat() {
         return rdf;
     }
+
+    public long getHitlistPage() {
```
Comment (henning-gerhardt) [inline Kitodo/src/main/java/de/sub/goobi/forms/ProzesskopieForm.java]: New public method without JavaDoc - please add missing JavaDoc.

File: Kitodo/src/main/java/de/sub/goobi/forms/ProzesskopieForm.java [hunk 2]
```diff
@@ -2013,4 +739,20 @@ public boolean isCalendarButtonShowing() {
...
+        return hitlistPage;
+    }
+
+    public void setHitlistPage(long hitlistPage) {
```
Comment (henning-gerhardt) [inline Kitodo/src/main/java/de/sub/goobi/forms/ProzesskopieForm.java]: New public method without JavaDoc - please add missing JavaDoc.

File: Kitodo/src/main/java/de/sub/goobi/forms/ProzesskopieForm.java [hunk 3]
```diff
@@ -2013,4 +739,20 @@ public boolean isCalendarButtonShowing() {
...
+        this.hitlistPage = hitlistPage;
+    }
+
+    public CopyProcess getCopyProcess() {
```
Comment (henning-gerhardt) [inline Kitodo/src/main/java/de/sub/goobi/forms/ProzesskopieForm.java]: New public method without JavaDoc - please add missing JavaDoc.

File: Kitodo/src/main/java/de/sub/goobi/forms/ProzesskopieForm.java [hunk 4]
```diff
@@ -2013,4 +739,20 @@ public boolean isCalendarButtonShowing() {
...
+        return copyProcess;
+    }
+
+    public void setCopyProcess(CopyProcess copyProcess) {
```
Comment (henning-gerhardt) [inline Kitodo/src/main/java/de/sub/goobi/forms/ProzesskopieForm.java]: New public method without JavaDoc - please add missing JavaDoc.

File: Kitodo/src/main/java/org/goobi/production/cli/helper/CopyProcess.java [hunk 5]
```diff
@@ -643,6 +590,282 @@ public Process createProcess(ImportObject io)
         return this.prozessKopie;
     }
 
+    public boolean createNewProcess()
```
Comment (henning-gerhardt) [inline Kitodo/src/main/java/org/goobi/production/cli/helper/CopyProcess.java]: New public method without JavaDoc - please add missing JavaDoc.

File: Kitodo/src/main/java/org/goobi/production/cli/helper/CopyProcess.java [hunk 6]
```diff
@@ -643,6 +590,282 @@ public Process createProcess(ImportObject io)
...
+        if (!successful) {
+            String message = "Metadata directory: " + baseProcessDirectory + "in path:"
+                    +  ConfigCore.getKitodoDataDirectory() + " was not created!";
+            logger.error(message);
+            Helper.setFehlerMeldung(message);
+            return false;
+        }
+
+        /*
+         * wenn noch keine RDF-Datei vorhanden ist (weil keine Opac-Abfrage
```
Comment (henning-gerhardt) [inline Kitodo/src/main/java/org/goobi/production/cli/helper/CopyProcess.java]: Please use an English comment.

File: Kitodo/src/main/java/org/goobi/production/cli/helper/CopyProcess.java [hunk 7]
```diff
@@ -643,6 +590,282 @@ public Process createProcess(ImportObject io)
...
+         * stattfand, dann jetzt eine anlegen
+         */
+        if (this.rdf == null) {
+            createNewFileformat();
+        }
+
+        /*
+         * wenn eine RDF-Konfiguration vorhanden ist (z.B. aus dem Opac-Import,
```
Comment (henning-gerhardt) [inline Kitodo/src/main/java/org/goobi/production/cli/helper/CopyProcess.java]: Please use an English comment.

File: Kitodo/src/main/java/org/goobi/production/cli/helper/CopyProcess.java [hunk 8]
```diff
@@ -643,6 +590,282 @@ public Process createProcess(ImportObject io)
...
+                    if (field.getDocstruct().equals("firstchild")) {
+                        try {
+                            tempStruct = this.rdf.getDigitalDocument().getLogicalDocStruct().getAllChildren()
+                                    .get(0);
+                        } catch (RuntimeException e) {
+                            logger.error(e.getMessage() + " The first child below the top structure could not be determined!");
+                        }
+                    }
+                    /*
+                     * falls topstruct und firstchild das Metadatum bekommen
```
Comment (henning-gerhardt) [inline Kitodo/src/main/java/org/goobi/production/cli/helper/CopyProcess.java]: Please use an English comment.

File: Kitodo/src/main/java/org/goobi/production/cli/helper/CopyProcess.java [hunk 9]
```diff
@@ -643,6 +590,282 @@ public Process createProcess(ImportObject io)
...
+                        try {
+                            tempChild = this.rdf.getDigitalDocument().getLogicalDocStruct().getAllChildren().get(0);
+                        } catch (RuntimeException e) {
+                            logger.error(e);
+                        }
+                    }
+                    if (field.getDocstruct().equals("boundbook")) {
+                        tempStruct = this.rdf.getDigitalDocument().getPhysicalDocStruct();
+                    }
+                    /* welches Metadatum */
```
Comment (henning-gerhardt) [inline Kitodo/src/main/java/org/goobi/production/cli/helper/CopyProcess.java]: Please use an English comment.

File: Kitodo/src/main/java/org/goobi/production/cli/helper/CopyProcess.java [hunk 10]
```diff
@@ -643,6 +590,282 @@ public Process createProcess(ImportObject io)
...
+                    try {
+                        /*
+                         * bis auf die Autoren alle additionals in die Metadaten
```
Comment (henning-gerhardt) [inline Kitodo/src/main/java/org/goobi/production/cli/helper/CopyProcess.java]: Please use an English comment.

File: Kitodo/src/main/java/org/goobi/production/cli/helper/CopyProcess.java [hunk 11]
```diff
@@ -643,6 +590,282 @@ public Process createProcess(ImportObject io)
...
+                        if (!field.getMetadata().equals("ListOfCreators")) {
+                            MetadataType mdt = UghHelper.getMetadataType(
+                                    serviceManager.getRulesetService().getPreferences(this.prozessKopie.getRuleset()),
+                                    field.getMetadata());
+                            Metadata metadata = UghHelper.getMetadata(tempStruct, mdt);
+                            if (metadata != null) {
+                                metadata.setValue(field.getValue());
+                            }
+                            /*
+                             * wenn dem Topstruct und dem Firstchild der Wert
```
Comment (henning-gerhardt) [inline Kitodo/src/main/java/org/goobi/production/cli/helper/CopyProcess.java]: Please use an English comment.

File: Kitodo/src/main/java/org/goobi/production/cli/helper/CopyProcess.java [hunk 12]
```diff
@@ -643,6 +590,282 @@ public Process createProcess(ImportObject io)
...
+                    } catch (Exception e) {
+                        Helper.setFehlerMeldung(e);
+                    }
+                }
+            }
+
+            updateMetadata();
+
+            /*
+             * Collectionen hinzufügen
```
Comment (henning-gerhardt) [inline Kitodo/src/main/java/org/goobi/production/cli/helper/CopyProcess.java]: Please use an English comment.

File: Kitodo/src/main/java/org/goobi/production/cli/helper/CopyProcess.java [hunk 13]
```diff
@@ -643,6 +590,282 @@ public Process createProcess(ImportObject io)
...
+             */
+            DocStruct colStruct = this.rdf.getDigitalDocument().getLogicalDocStruct();
+            try {
+                addCollections(colStruct);
+                /*
+                 * falls ein erstes Kind vorhanden ist, sind die Collectionen
```
Comment (henning-gerhardt) [inline Kitodo/src/main/java/org/goobi/production/cli/helper/CopyProcess.java]: Please use an English comment.

File: Kitodo/src/main/java/org/goobi/production/cli/helper/CopyProcess.java [hunk 14]
```diff
@@ -643,6 +590,282 @@ public Process createProcess(ImportObject io)
...
+                 * dafür
+                 */
+                colStruct = colStruct.getAllChildren().get(0);
+                addCollections(colStruct);
+            } catch (RuntimeException e) {
+                logger.error(e.getMessage() + " The first child below the top structure could not be determined!");
+            }
+
+            /*
+             * Imagepfad hinzufügen (evtl. vorhandene zunächst löschen)
```
Comment (henning-gerhardt) [inline Kitodo/src/main/java/org/goobi/production/cli/helper/CopyProcess.java]: Please use an English comment.

File: Kitodo/src/main/java/org/goobi/production/cli/helper/CopyProcess.java [hunk 15]
```diff
@@ -643,6 +590,282 @@ public Process createProcess(ImportObject io)
...
+                if (SystemUtils.IS_OS_WINDOWS) {
+                    newMetadata.setValue("file:/" + serviceManager.getFileService().getImagesDirectory(this.prozessKopie)
+                            + this.prozessKopie.getTitle().trim() + DIRECTORY_SUFFIX);
+                } else {
+                    newMetadata.setValue("file://" + serviceManager.getFileService().getImagesDirectory(this.prozessKopie)
+                            + this.prozessKopie.getTitle().trim() + DIRECTORY_SUFFIX);
+                }
+                this.rdf.getDigitalDocument().getPhysicalDocStruct().addMetadata(newMetadata);
+
+                /* Rdf-File schreiben */
```
Comment (henning-gerhardt) [inline Kitodo/src/main/java/org/goobi/production/cli/helper/CopyProcess.java]: Please use an English comment.

File: Kitodo/src/main/java/org/goobi/production/cli/helper/CopyProcess.java [hunk 16]
```diff
@@ -643,6 +590,282 @@ public Process createProcess(ImportObject io)
...
+                serviceManager.getFileService().writeMetadataFile(this.rdf, this.prozessKopie);
+
+                /*
+                 * soll der Prozess als Vorlage verwendet werden?
```
Comment (henning-gerhardt) [inline Kitodo/src/main/java/org/goobi/production/cli/helper/CopyProcess.java]: Please use an English comment.

File: Kitodo/src/main/java/org/goobi/production/cli/helper/CopyProcess.java [hunk 17]
```diff
@@ -701,12 +952,11 @@ private void removeCollections(DocStruct colStruct) {
...
         } catch (UghHelperException | DocStructHasNoTypeException e) {
             Helper.setFehlerMeldung(e.getMessage(), "");
-            e.printStackTrace();
+            logger.error(e);
         }
     }
 
-    @Override
-    public void createNewFileformat() {
+    /*public void createNewFileformat() {
```
Comment (henning-gerhardt) [inline Kitodo/src/main/java/org/goobi/production/cli/helper/CopyProcess.java]: Please remove unused code and do not comment it out.

File: Kitodo/src/main/java/org/goobi/production/cli/helper/CopyProcess.java [hunk 18]
```diff
@@ -1112,48 +1464,53 @@ private String calcProcessTitleCheck(String fieldName, String fieldvalue) {
...
+        // Documentname ist im allgemeinen = Prozesstitel
         this.tifHeaderDocumentName = this.prozessKopie.getTitle();
         this.tifHeaderImageDescription = new StringBuilder("");
-        /*
-         * Imagedescription
-         */
+        // image description
         StringTokenizer tokenizer = new StringTokenizer(tifDefinition, "+");
-        /* jetzt den Tiffheader parsen */
+        // jetzt den Tiffheader parsen
```
Comment (henning-gerhardt) [inline Kitodo/src/main/java/org/goobi/production/cli/helper/CopyProcess.java]: Please use an English comment.

File: Kitodo/src/main/java/org/goobi/production/cli/helper/CopyProcess.java [hunk 19]
```diff
@@ -1112,48 +1464,53 @@ private String calcProcessTitleCheck(String fieldName, String fieldvalue) {
...
-         * evtuelle Ersetzungen
-         */
+        // possible replacements
         tifDefinition = tifDefinition.replaceAll("\\[\\[", "<");
         tifDefinition = tifDefinition.replaceAll("\\]\\]", ">");
 
-        /*
-         * Documentname ist im allgemeinen = Prozesstitel
-         */
+        // Documentname ist im allgemeinen = Prozesstitel
```
Comment (henning-gerhardt) [inline Kitodo/src/main/java/org/goobi/production/cli/helper/CopyProcess.java]: Please use an English comment.

File: Kitodo/src/main/java/org/goobi/production/cli/helper/CopyProcess.java [hunk 20]
```diff
@@ -1112,48 +1464,53 @@ private String calcProcessTitleCheck(String fieldName, String fieldvalue) {
...
         StringTokenizer tokenizer = new StringTokenizer(tifDefinition, "+");
-        /* jetzt den Tiffheader parsen */
+        // jetzt den Tiffheader parsen
+        String title = "";
         while (tokenizer.hasMoreTokens()) {
-            String string = tokenizer.nextToken();
+            String tokenString = tokenizer.nextToken();
             /*
-             * wenn der String mit ' anfängt und mit ' endet, dann den Inhalt so übernehmen
+             * wenn der String mit ' anfängt und mit ' endet, dann den Inhalt so
```
Comment (henning-gerhardt) [inline Kitodo/src/main/java/org/goobi/production/cli/helper/CopyProcess.java]: Please use an English comment.

File: Kitodo/src/main/java/org/goobi/production/cli/helper/CopyProcess.java [hunk 21]
```diff
@@ -1112,48 +1464,53 @@ private String calcProcessTitleCheck(String fieldName, String fieldvalue) {
...
              */
-            if (string.startsWith("'") && string.endsWith("'") && string.length() > 2) {
-                this.tifHeaderImageDescription.append(string.substring(1, string.length() - 1));
-            } else if (string.equals("$Doctype")) {
-
-                this.tifHeaderImageDescription.append(this.docType);
+            if (tokenString.startsWith("'") && tokenString.endsWith("'") && tokenString.length() > 2) {
+                this.tifHeaderImageDescription.append(tokenString.substring(1, tokenString.length() - 1));
+            } else if (tokenString.equals("$Doctype")) {
+                /* wenn der Doctype angegeben werden soll */
```
Comment (henning-gerhardt) [inline Kitodo/src/main/java/org/goobi/production/cli/helper/CopyProcess.java]: Please use an English comment.

File: Kitodo/src/main/java/org/goobi/production/cli/helper/CopyProcess.java [hunk 22]
```diff
@@ -1240,4 +1610,219 @@ public void setMetadataFile(URI mdFile) {
...
+                titel = "";
+            }
+            if (isdoctype == null) {
+                isdoctype = "";
+            }
+            if (isnotdoctype == null) {
+                isnotdoctype = "";
+            }
+
+            /* wenn nix angegeben wurde, dann anzeigen */
```
Comment (henning-gerhardt) [inline Kitodo/src/main/java/org/goobi/production/cli/helper/CopyProcess.java]: Please use an English comment.

File: Kitodo/src/main/java/org/goobi/production/cli/helper/CopyProcess.java [hunk 23]
```diff
@@ -1240,4 +1610,219 @@ public void setMetadataFile(URI mdFile) {
...
+            if (isdoctype.equals("") && isnotdoctype.equals("")) {
+                titeldefinition = titel;
+                break;
+            }
+
+            /* wenn beides angegeben wurde */
```
Comment (henning-gerhardt) [inline Kitodo/src/main/java/org/goobi/production/cli/helper/CopyProcess.java]: Please use an English comment.

File: Kitodo/src/main/java/org/goobi/production/cli/helper/CopyProcess.java [hunk 24]
```diff
@@ -1240,4 +1610,219 @@ public void setMetadataFile(URI mdFile) {
...
+            if (!isdoctype.equals("") && !isnotdoctype.equals("")
+                    && StringUtils.containsIgnoreCase(isdoctype, this.docType)
+                    && !StringUtils.containsIgnoreCase(isnotdoctype, this.docType)) {
+                titeldefinition = titel;
+                break;
+            }
+
+            /* wenn nur pflicht angegeben wurde */
```
Comment (henning-gerhardt) [inline Kitodo/src/main/java/org/goobi/production/cli/helper/CopyProcess.java]: Please use an English comment.

File: Kitodo/src/main/java/org/goobi/production/cli/helper/CopyProcess.java [hunk 25]
```diff
@@ -1240,4 +1610,219 @@ public void setMetadataFile(URI mdFile) {
...
+            if (isnotdoctype.equals("") && StringUtils.containsIgnoreCase(isdoctype, this.docType)) {
+                titeldefinition = titel;
+                break;
+            }
+            /* wenn nur "darf nicht" angegeben wurde */
```
Comment (henning-gerhardt) [inline Kitodo/src/main/java/org/goobi/production/cli/helper/CopyProcess.java]: Please use an English comment.

File: Kitodo/src/main/java/org/goobi/production/cli/helper/CopyProcess.java [hunk 26]
```diff
@@ -1240,4 +1610,219 @@ public void setMetadataFile(URI mdFile) {
...
+            if (isdoctype.equals("") && !StringUtils.containsIgnoreCase(isnotdoctype, this.docType)) {
+                titeldefinition = titel;
+                break;
+            }
+        }
+
+        StringTokenizer tokenizer = new StringTokenizer(titeldefinition, "+");
+        /* jetzt den Bandtitel parsen */
```
Comment (henning-gerhardt) [inline Kitodo/src/main/java/org/goobi/production/cli/helper/CopyProcess.java]: Please use an English comment.

File: Kitodo/src/main/java/org/goobi/production/cli/helper/CopyProcess.java [hunk 27]
```diff
@@ -1240,4 +1610,219 @@ public void setMetadataFile(URI mdFile) {
...
+        while (tokenizer.hasMoreTokens()) {
+            String myString = tokenizer.nextToken();
+            /*
+             * wenn der String mit ' anfängt und mit ' endet, dann den Inhalt so
```
Comment (henning-gerhardt) [inline Kitodo/src/main/java/org/goobi/production/cli/helper/CopyProcess.java]: Please use an English comment.

File: Kitodo/src/main/java/org/goobi/production/cli/helper/CopyProcess.java [hunk 28]
```diff
@@ -1240,4 +1610,219 @@ public void setMetadataFile(URI mdFile) {
...
+                 * resolve strings beginning with # from generic fields
+                 */
+                if (genericFields != null) {
+                    String genericValue = genericFields.get(myString);
+                    if (genericValue != null) {
+                        newTitle.append(genericValue);
+                    }
+                }
+            } else {
+                /* andernfalls den string als Feldnamen auswerten */
```
Comment (henning-gerhardt) [inline Kitodo/src/main/java/org/goobi/production/cli/helper/CopyProcess.java]: Please use an English comment.

File: Kitodo/src/main/java/org/goobi/production/cli/helper/CopyProcess.java [hunk 29]
```diff
@@ -1240,4 +1610,219 @@ public void setMetadataFile(URI mdFile) {
...
+                for (AdditionalField additionalField : this.additionalFields) {
+                    /*
+                     * wenn es das ATS oder TSL-Feld ist, dann den berechneten
```
Comment (henning-gerhardt) [inline Kitodo/src/main/java/org/goobi/production/cli/helper/CopyProcess.java]: Please use an English comment.

File: Kitodo/src/main/java/org/goobi/production/cli/helper/CopyProcess.java [hunk 30]
```diff
@@ -1240,4 +1610,219 @@ public void setMetadataFile(URI mdFile) {
...
+                    if ((additionalField.getTitle().equals("ATS") || additionalField.getTitle().equals("TSL"))
+                            && additionalField.getShowDependingOnDoctype()
+                            && (additionalField.getValue() == null || additionalField.getValue().equals(""))) {
+                        if (atstsl == null || atstsl.length() == 0) {
+                            atstsl = createAtstsl(currentTitle, currentAuthors);
+                        }
+                        additionalField.setValue(this.atstsl);
+                    }
+
+                    /* den Inhalt zum Titel hinzufügen */
```
Comment (henning-gerhardt) [inline Kitodo/src/main/java/org/goobi/production/cli/helper/CopyProcess.java]: Please use an English comment.

File: Kitodo/src/main/java/org/goobi/production/cli/helper/CopyProcess.java [hunk 31]
```diff
@@ -1240,4 +1610,219 @@ public void setMetadataFile(URI mdFile) {
...
+        prozessKopie.setTitle(filteredTitle);
+        calculateTiffHeader();
+        return filteredTitle;
+    }
+
+    private String calculateProcessTitleCheck(String inFeldName, String inFeldWert) {
+        String rueckgabe = inFeldWert;
+
+        /*
+         * Bandnummer
```
Comment (henning-gerhardt) [inline Kitodo/src/main/java/org/goobi/production/cli/helper/CopyProcess.java]: Please use an English comment.

File: Kitodo/src/main/java/org/kitodo/helper/SelectableHit.java [hunk 32]
```diff
@@ -0,0 +1,122 @@
...
+     */
+    public boolean isError() {
+        return hit == null;
+    }
+
+    /**
+     * The function selectClick() is called if the user clicks on a catalogue hit
+     * summary in order to import it into Production.
+     *
+     * @return always "", indicating to Faces to stay on that page
```
Comment (henning-gerhardt) [inline Kitodo/src/main/java/org/kitodo/helper/SelectableHit.java]: There is a mismatch between return value description and returned value.

File: Kitodo/src/main/java/de/sub/goobi/forms/ProzesskopieForm.java [hunk 33]
```diff
@@ -428,7 +192,7 @@ public String evaluateOpac() {
         long timeout = CataloguePlugin.getTimeout();
         try {
             clearValues();
-            readProjectConfigs();
+            //readProjectConfigs();
```
[the remark this answers is not in the document]
Comment (Beacze) [reply Kitodo/src/main/java/de/sub/goobi/forms/ProzesskopieForm.java]: I'm not sure if it is needed here...

**Comment by henning-gerhardt** [comment]
Further review not possible as it is currently not possible to create new processes from a template to get data from a catalogue OPAC.

**Comment by Beacze** [comment]
Obsolete

**Pull request closed without merging**